cli/src/semdep/parse_lockfile.py

```diff
@@ -661,14 +661,43 @@ def parse_manifest(text: str) -> Set[str]:
         )
 
 
+def parse_pom_tree(tree_str: str, _: Optional[str]) -> Iterator[FoundDependency]:
+    def package_index(line: str) -> int:
+        i = 0
+        while line[i] in ["+", "-", " ", "\\"]:
+            i += 1
+        return i
+
+    def depth(package_index: int) -> int:
+        return package_index // 3 + 1
+
+    lines = tree_str.split("\n")
+    deps = [l for l in lines[1:] if l]  # Filter any empty lines
+    for i, dep in enumerate(deps):
+        j = package_index(dep)
```

Comment:
You can use .lstrip()?

Reply:
Ah I see you needed the index later

Reply:
It would have to strip multiple different characters

Reply:
lstrip does "abc123".lstrip("cab") returns "123"